src/Action.ts

```diff
@@ -1,141 +1,157 @@
-            let getResponse: ReleaseByTagResponse
-            try {
-                getResponse = await this.releases.getByTag(this.inputs.tag)
-            } catch (error: any) {
-                return await this.checkForMissingReleaseError(error)
-            }
-            
-            // Fail if this isn't an unreleased release & updateOnlyUnreleased is enabled.
-            this.releaseValidator.validateReleaseUpdate(getResponse.data)
-
-            return await this.updateRelease(getResponse.data.id)
-        } else {
-            return await this.createRelease()
-        }
-    }
-
-    private async checkForMissingReleaseError(error: Error): Promise<CreateOrUpdateReleaseResponse> {
-        if (Action.noPublishedRelease(error)) {
-            return await this.updateDraftOrCreateRelease()
-        } else {
-            throw error
-        }
-    }
-
-    private async updateRelease(id: number): Promise<UpdateReleaseResponse> {
-        return await this.releases.update(
-            id,
-            this.inputs.tag,
-            this.inputs.updatedReleaseBody,
-            this.inputs.commit,
-            this.inputs.discussionCategory,
-            this.inputs.updatedDraft,
-            this.inputs.makeLatest,
-            this.inputs.updatedReleaseName,
-            this.inputs.updatedPrerelease
-        )
-    }
-
-    private static noPublishedRelease(error: any): boolean {
-        const githubError = new GithubError(error)
-        return githubError.status == 404
-    }
-
-    private async updateDraftOrCreateRelease(): Promise<CreateReleaseResponse | UpdateReleaseResponse> {
-        const draftReleaseId = await this.findMatchingDraftReleaseId()
-        if (draftReleaseId) {
-            return await this.updateRelease(draftReleaseId)
-        } else {
-            return await this.createRelease()
-        }
-    }
-
-    private async findMatchingDraftReleaseId(): Promise<number | undefined> {
-        const tag = this.inputs.tag
-        const response = await this.releases.listReleases()
-        const releases = response.data
-        const draftRelease = releases.find(release => release.draft && release.tag_name == tag)
-
-        return draftRelease?.id
-    }
-
-    private async createRelease(): Promise<CreateReleaseResponse> {
-        return await this.releases.create(
-            this.inputs.tag,
-            this.inputs.createdReleaseBody,
-            this.inputs.commit,
-            this.inputs.discussionCategory,
-            this.inputs.createdDraft,
-            this.inputs.generateReleaseNotes,
-            this.inputs.makeLatest,
-            this.inputs.createdReleaseName,
-            this.inputs.createdPrerelease
-        )
-    }
+	private inputs: Inputs;
+	private outputs: Outputs;
+	private releases: Releases;
+	private uploader: ArtifactUploader;
+	private artifactDestroyer: ArtifactDestroyer;
+	private skipper: ActionSkipper;
+
+	private releaseValidator: ReleaseValidator;
```

Comment:
Seems like your IDE reformatted this file pretty aggressively. I might re-format it again before we merge down.